app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    previous_args = {}
+
+    def inner(*args) -> Any:
+        if all([
+            isinstance(arg, (tuple, str, int, float, bool))
+            for arg in args
+        ]):
+            if args in previous_args.keys():
+                print("Getting from cache")
+                return previous_args[args]
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it.